source/solvers/navier_stokes_base.cc

```diff
@@ -865,6 +865,14 @@ NavierStokesBase<dim, VectorType, DofsType>::refine_mesh_kelly()
   const FEValuesExtractors::Scalar pressure(dim);
   auto &                           present_solution = this->present_solution;
 
+  // Global flags
+  // Their dimension is consistent with the dimension returned by
+  // save_refine_flags(), in order to be able to use load_refine_flags()
+  std::vector<bool> global_refine_flags(dim * tria.n_active_cells(), false);
+  std::vector<bool> global_coarsen_flags(dim * tria.n_active_cells(), false);
+
+  bool first_variable(true);
```

Comment:
What you could also do is not give a size to the vector and use their size to check if it's been filled or no, this would save you the use of another variable, but it's maybe more readable in the current fashion.

Reply:
Yes I tried not using another variable at first but I liked that using a simple boolean made this extra clear

Reply:
Cool with me